lib/repl.js

```diff
@@ -521,7 +521,7 @@ function REPLServer(prompt,
           if (options === null || typeof options !== 'object') {
             throw new ERR_INVALID_ARG_TYPE('options', 'Object', options);
           }
-          return Object.assign(writer.options, options);
+          return { ...writer.options, ...options };
```

Comment:
This isn't correct, this is a setter that should be saving options to the defaults. This is now just returning a new object with the properties combined, leaving the defaults intact.

Reply:
Updated